R/database.R

```diff
@@ -1715,98 +1715,154 @@ get_sample_cn_segments = function(this_sample_id,
+#'                                        remove_chr_prefix = TRUE, 
+#'                                        from_flatfile = TRUE, 
+#'                                        streamlined = FALSE) 
+#'
+#'
+get_cn_segments = function(region,
+                           chromosome,
                            qstart,
                            qend,
-                           region,
-                           with_chr_prefix = FALSE,
+                           projection = "grch37",
+                           this_seq_type = "genome",
+                           remove_chr_prefix = FALSE,
                            streamlined = FALSE,
-                           from_flatfile = FALSE){
+                           from_flatfile = TRUE){
+  
+  #checks
   remote_session = check_remote_configuration()
-  db = config::get("database_name")
-  table_name = config::get("results_tables")$copy_number
-  table_name_unmatched = config::get("results_tables")$copy_number_unmatched
+  
+  #check seq type and return a message if anything besides "genome" is called. To be updated once capture samples have been processed through CNV protocols.
+  if(this_seq_type!="genome"){
+    stop("Currently, only genome samples are available for this function. Please select a valid seq type (i.e genome). Compatibility for capture samples will be added soon...")
+  }
+  
+  #get wildcards from this_seq_type (lazy)
+  seq_type = this_seq_type
+  
+  #perform wrangling on the region to have it in the correct format. 
   if(!missing(region)){
     region = gsub(",", "", region)
-    #format is chr6:37060224-37151701
     split_chunks = unlist(strsplit(region, ":"))
     chromosome = split_chunks[1]
     startend = unlist(strsplit(split_chunks[2], "-"))
     qstart = startend[1]
     qend = startend[2]
   }
-  if(grepl("chr", chromosome)){
+  
+  #deal with chr prefixes for region, based on selected genome projection.
+  if(projection == "grch37"){
+    if(grepl("chr", chromosome)){
+      chromosome = gsub("chr", "", chromosome)
+    }
   }else{
-    chromosome = paste0("chr", chromosome)
+    if(!grepl("chr", chromosome)){
+      chromosome = paste0("chr", chromosome)
+    }
   }
-
-  #chr prefix the query chromosome to match how it's stored in the table.
-  #This isn't yet standardized in the db so it's just a workaround "for now".
+  
+  #sanity check type of qstart en qend
+  if(!is.numeric(qstart)){
+    qstart = as.numeric(qstart)
+  }
+  
+  if(!is.numeric(qend)){
+    qend = as.numeric(qend)
+  }
+  
   if(from_flatfile){
-    base_dir = config::get(config="default")$project_base
-
-    unmatched_path = config::get()$results_directories$controlfreec
-
-    #separated by which genome the sample was aligned to
-    unmatched_hg38_path = paste0(unmatched_path, "from--genome--hg38/")
-    unmatched_grch37_path = paste0(unmatched_path, "from--genome--grch37/")
+    cnv_flatfile_template = config::get("results_flatfiles")$cnv_combined$icgc_dart
```

Comment:
We might want to check the user permission on this file here if the user is outside the morinlab, then default to gambl and give a warning. This way, it will not error out for other GSC users who do not have ICGC permissions